### PR TITLE
Improve dump messages for jump operations

### PR DESCRIPTION
**Pull request by rerobika**
Signed-off-by: Robert Fancsik <frobert@inf.u-szeged.hu>

### CODE REVIEW

**Comment by rerobika** [comment]
To demonstrate the affect of the PR in a single example:

``` js
if (true) {
  print ("true");
} else {
  print ("false");
}
```
Bytecode dump before:

```
dumpBytecode  (1:1)>>>>>>>>>>>>>>>>>>>>>>
register info.. (stack variable size(1)) [X,X,X,`this`,(4,true),]
0               jump if false r4 -> 1438328752 | JumpIfFalse (line: 12:4)
48              get global object r1 <- global.print | GetGlobalObject (line: 13:2)
112             load r2 <- "true" | LoadLiteral (line: 13:9)
160             call r0 <- r1(r2-r3) | CallFunction (line: 13:2)
208             jump 1438328912 | Jump (line: 12:4)
256             get global object r1 <- global.print | GetGlobalObject (line: 15:2)
320             load r2 <- "false" | LoadLiteral (line: 15:9)
368             call r0 <- r1(r2-r3) | CallFunction (line: 15:2)
416             end | End (line: 12:1)
dumpBytecode...<<<<<<<<<<<<<<<<<<<<<<
```

And after:
```
dumpBytecode  (1:1)>>>>>>>>>>>>>>>>>>>>>>
register info.. (stack variable size(1)) [X,X,X,`this`,(4,true),]
0               jump if false r4 -> 256 | JumpIfFalse (line: 12:4)
48              get global object r1 <- global.print | GetGlobalObject (line: 13:2)
112             load r2 <- "true" | LoadLiteral (line: 13:9)
160             call r0 <- r1(r2-r3) | CallFunction (line: 13:2)
208             jump 416 | Jump (line: 12:4)
256             get global object r1 <- global.print | GetGlobalObject (line: 15:2)
320             load r2 <- "false" | LoadLiteral (line: 15:9)
368             call r0 <- r1(r2-r3) | CallFunction (line: 15:2)
416             end | End (line: 12:1)
dumpBytecode...<<<<<<<<<<<<<<<<<<<<<<
```

File: src/interpreter/ByteCode.h [hunk 1]
```diff
@@ -188,7 +188,8 @@ class ByteCode {
     ByteCodeLOC m_loc;
     Opcode m_orgOpcode;
 
-    void dumpCode(size_t pos);
+    void dumpCode(size_t pos, char* code);
+    int dumpJumpPosition(size_t pos, char* code);
```
Comment (clover2123) [inline src/interpreter/ByteCode.h]: Its just my suggestion. Could you change each char type pointer parameter to pointer to constant char (`char*` -> `const char*`) ?

File: src/interpreter/ByteCode.cpp [hunk 2]
```diff
@@ -43,15 +43,15 @@ OpcodeTable::OpcodeTable()
 }
 
 #ifndef NDEBUG
-void ByteCode::dumpCode(size_t pos)
+void ByteCode::dumpCode(size_t pos, char* code)
```
Comment (zherczeg) [inline src/interpreter/ByteCode.cpp]: I would use `byteCode` or `byteCodeStart`.

File: src/interpreter/ByteCodeGenerator.cpp [hunk 3]
```diff
@@ -547,18 +547,18 @@ ByteCodeBlock* ByteCodeGenerator::generateByteCode(Context* c, InterpretedCodeBl
...
         while (idx < end) {
-            ByteCode* currentCode = (ByteCode*)(code + idx);
+            ByteCode* currentCode = (ByteCode*)(byteCode + idx);
 
             Opcode opcode = currentCode->m_orgOpcode;
             switch (opcode) {
 #define DUMP_BYTE_CODE(code, pushCount, popCount) \
     case code##Opcode:                            \
-        currentCode->dumpCode(idx);               \
+        currentCode->dumpCode(idx, byteCode);     \
```
Comment (zherczeg) [inline src/interpreter/ByteCodeGenerator.cpp]: Maybe this switch could be deleted.

**Comment by rerobika** [comment]
@clover2123 @zherczeg  Thanks for the reviews, the PR is updated!

**Pull request merged**